app/models/user.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class User < ApplicationRecord
+  validates :username, presence: true, uniqueness: true
+  before_save { email.downcase! }
+  before_save { username.downcase! }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i.freeze
+  validates :email, presence: true, length: { maximum: 255 }, format: { with: VALID_EMAIL_REGEX },
+                    uniqueness: { case_sensitive: false }
+  has_many :events, foreign_key: 'creator_id'
+  has_many :event_attendances, foreign_key: :event_attendee_id
+  has_many :attended_event, through: :event_attendances, source: :attendee_event
+  has_many :invitations, foreign_key: :invitation_user_id
+  has_many :event_invitations, through: :invitations, source: :invitation_event
+
+  def previous_events
+    attended_event.where('date <= ?', Time.now)
+  end
+
+  def upcoming_events
+    attended_event.where('date > ?', Time.now)
+  end
```

Comment:
Guys you don't need theses methods: `previos_events` and `upcoming_events` you can get that information from the association and the scope you already have in the Event model, something like, `current_user.attended_event.upcoming`. Also, `attended_event` should be plural because you are using a `has_many` association.